bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/TreeBuilder.java

```diff
@@ -106,22 +132,16 @@ private RegExpTerm readRegTerm() {
                 reAtom = readRegCapturingGroups();
                 break;
             default:
-                // Read chars in ReLiteralChar, . or ReEscape.
-                reAtom = readRegChars();
+                // Here the token is a syntax char, which is invalid. A syntax char should have a backslash prior
+                // to the token.
```

Comment:
What if this could be re-written as this?
```suggestion
                // Here the token is a syntax char, which is invalid. Syntax char tokens should be 
                // preceded by backslashes.
```